File: 3611-construct-the-minimum-bitwise-array-ii/construct-the-minimum-bitwise-array-ii.py
class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        ans = []
        
        for val in nums:
            if val == 2:
                ans.append(-1)
            else:
                trailing_ones = 0
                temp = val
                
                # Count trailing 1s
                while (temp & 1) == 1:
                    trailing_ones += 1
                    temp >>= 1
                
                # Flip the MSB of the trailing ones sequence
                ans.append(val - (1 << (trailing_ones - 1)))
                
        return ans

File: 3611-construct-the-minimum-bitwise-array-ii/construct-the-minimum-bitwise-array-ii.cpp
class Solution {
public:
    vector<int> minBitwiseArray(vector<int>& nums) {
        vector<int> ans;
        
        for (int val : nums) {
            if (val == 2) {
                ans.push_back(-1);
            } else {
                int trailingOnes = 0;
                int temp = val;
                
                // Count trailing 1s
                while ((temp & 1) == 1) {
                    trailingOnes++;
                    temp >>= 1;
                }
                
                // Flip the MSB of the trailing ones sequence
                ans.push_back(val - (1 << (trailingOnes - 1)));
            }
        }
        return ans;
    }
};